apps/modernization-ui/src/design-system/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,36 @@
+import { Pagination as PaginationUswds } from '@trussworks/react-uswds';
+import styles from './pagination.module.scss';
+import { RangeToggle } from 'components/Table/RangeToggle/RangeToggle';
+import { useEffect } from 'react';
+import { usePage } from 'page';
+
+type Props = {
+    total: number;
+    pageSize: number;
+};
+
+export const Pagination = ({ total, pageSize }: Props) => {
+    const { ready, page } = usePage();
+
+    useEffect(() => {
+        ready(total ?? 0, page.current);
+    }, [total, page]);
+
+    return (
+        <div className={styles.pagination}>
+            <div className={styles.range}>
+                Showing <RangeToggle /> of <span id="totalRowCount">{total}</span>
+            </div>
+            <div className={styles.pages}>
+                <PaginationUswds
+                    totalPages={Math.ceil(total / pageSize)}
+                    currentPage={page.current}
+                    pathname={``}
+                    onClickNext={() => ready(total, page.current + 1)}
```

Comment:
issue:  `ready` is the function used to update the Page Sate with the new total and page size, the page is ready for display.  The `request` function should be used here because you are requesting to see page X